### PR TITLE
[spark] update for append table

### PR DESCRIPTION
**Pull request by YannByron**
### Purpose

https://github.com/apache/paimon/issues/3057

### CODE REVIEW

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/RowLevelOp.scala [hunk 1]
```diff
@@ -18,28 +18,63 @@
 
 package org.apache.paimon.spark.catalyst.analysis
 
-import org.apache.paimon.CoreOptions.MergeEngine
+import org.apache.paimon.AppendOnlyFileStore
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/RowLevelOp.scala:21]: Unused.

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/RowLevelOp.scala [hunk 2]
```diff
@@ -18,28 +18,63 @@
...
-  val supportedMergeEngine: Seq[MergeEngine]
+
+  val name: String = this.getClass.getSimpleName.stripSuffix("$")
+
+  protected val supportedMergeEngine: Seq[MergeEngine]
+
+  protected val supportedTable: Seq[String]
+
+  def checkValidity(table: Table): Unit = {
+    if (!supportedTable.contains(table.getClass.getSimpleName)) {
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/RowLevelOp.scala:36]: Just check it should be a `FileStoreTable`? And with or without primary key? It is better to just rely on public API.

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/RowLevelOp.scala [hunk 3]
```diff
@@ -18,28 +18,63 @@
...
+    }
+  }
 }
 
 case object Delete extends RowLevelOp {
-  override def toString: String = "delete"
 
   override val supportedMergeEngine: Seq[MergeEngine] = Seq(MergeEngine.DEDUPLICATE)
+
+  override val supportedTable: Seq[String] = Seq(PrimaryKeyTable)
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/RowLevelOp.scala:53]: `supportAppendTable`?

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/SparkDataFileMeta.scala [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    partition: BinaryRow,
+    bucket: Int,
+    totalBuckets: Int,
+    dataFileMeta: DataFileMeta
+) {
+
+  def relativePath(fileStorePathFactory: FileStorePathFactory): String = {
+    val partitionPath = fileStorePathFactory.getPartitionString(partition)
+    if (partitionPath.isEmpty) {
+      FileStorePathFactory.BUCKET_PATH_PREFIX + bucket + "/" +
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/SparkDataFileMeta.scala:38]: Use `FileStorePathFactory.bucketPath`.

Comment (YannByron) [reply paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/SparkDataFileMeta.scala:38]: Here, we don't need the root path prefix.

Comment (YannByron) [reply paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/SparkDataFileMeta.scala:38]: Or, I can add a method in `FileStorePathFactory` which return the relative bucket path.

**Pull request merged**